app/usecases/gdpr.py

```diff
@@ -0,0 +1,143 @@
+        for table, csv in (await generate_csvs(user_id)).items():
+            zip.writestr(f"data/{table}.csv", csv)
+
+        # Add the user avatar to the zip archive
+        avatars = glob.glob(str(app.utils.DATA_PATH / "avatars") + f"/{user_id}.*")
+        for avatar in avatars:
+            with open(avatar, "rb") as file:
+                zip.writestr(f"{os.path.split(avatar)[-1]}", file.read())
+
+        # Add the chat logs from .data/logs/chat.log
+        chatlog = StringIO()
+        file = open(app.utils.DATA_PATH / "logs/chat.log")
+        while True:
+            line = file.readline()
+            if not line:
+                break
+
+            matches = re.findall(
+                regexes.CHATLOG_USER_ID,
+                ":".join(line.split(":")[:3]),
+            )
+
+            if len(matches) > 0 and matches[0] == str(user_id):
+                chatlog.write(line)
+            if len(matches) > 1 and matches[1] == str(user_id):
+                chatlog.write(line)
+
+        chatlog.seek(0)
+        zip.writestr("chat.log", chatlog.getvalue())
+
+        # Get the score ids of all passed scores of the user and add their replays to the archive
+        # score_ids = await app.state.services.database.fetch_all(
+        #    "SELECT id FROM scores WHERE userid = :userid AND status > 0",
+        #    {"userid": user_id}
+        # )
+        # for id in (x["id"] for x in score_ids):
+        #    with open(app.utils.DATA_PATH / f"osr/{id}.osr", "rb") as file:
+        #        zip.writestr(f"replays/{id}.osr", file.read())
+
+    output.seek(0)
+    return output
+
+
+async def send_gdpr_email(player: Player) -> None:
+    """Sends an email containing the GDPR data to the specified user."""
+
+    zip = (await generate_zip_archive(player.id)).read()
+    attachment = app.usecases.email.get_file_attachment(f"gdpr_{player.id}.zip", zip)
+
+    message_base64 = "PGxpbmsgaHJlZj0iaHR0cHM6Ly9mb250cy5nb29nbGVhcGlzLmNvbS9jc3M/ZmFtaWx5PVJvYm90bzo0MDAsNzAwIiByZWw9InN0eWxlc2hlZXQiIHR5cGU9InRleHQvY3NzIj48dGFibGUgcm9sZT0icHJlc2VudGF0aW9uIiB3aWR0aD0iMTAwJSIgY2VsbHNwYWNpbmc9IjAiIGNlbGxwYWRkaW5nPSIwIj4gPHRib2R5PiA8dHI+IDx0ZD4gPGgxIHN0eWxlPSJtYXJnaW4tYm90dG9tOiAxMHB4OyB0ZXh0LWFsaWduOiBjZW50ZXI7IGZvbnQtZmFtaWx5OiBSb2JvdG8sIFRhaG9tYSwgc2Fucy1zZXJpZjsgZm9udC1zaXplOiA2NXB4OyI+R0RQUiBSZXF1ZXN0PC9oMT4gPC90ZD48L3RyPjx0cj4gPHRkPiA8aDIgc3R5bGU9Im1hcmdpbi10b3A6IDBweDsgZm9udC1mYW1pbHk6IFJvYm90bywgVGFob21hLCBzYW5zLXNlcmlmOyBmb250LXNpemU6IDI0cHg7IHRleHQtYWxpZ246IGNlbnRlcjsiPkhlbGxvIFtOQU1FXSE8YnIvPllvdXIgR0RQUiBkYXRhIHBhY2thZ2UgaXMgcmVhZHkuPGJyLz5QbGVhc2UgdmlldyB0aGUgYXR0YWNobWVudHMgdG8gZG93bmxvYWQ8YnIvPnRoZSB1bmNvbXByZXNzZWQgWklQIGFyY2hpdmUuPC9oMj4gPC90ZD48L3RyPjx0cj4gPHRkPiA8aDMgc3R5bGU9ImZvbnQtZmFtaWx5OiBSb2JvdG8sIFRhaG9tYSwgc2Fucy1zZXJpZjsgZm9udC1zaXplOiAxOHB4OyBjb2xvcjogIzk1YTVhNjsgdGV4dC1hbGlnbjogY2VudGVyOyI+U2VudCBieSBiYW5jaG8ucHkgdmlhIFtET01BSU5dPC9oMz4gPC90ZD48L3RyPjwvdGJvZHk+PC90YWJsZT4="
```

Comment:
Yeah I couldn't think of a better way to do this but I've seen base64 strings used a lot in the past for such things. If someone wants to modify it they can just decode it and done. Unless you have a better idea, I wanted to avoid creating a separate template file

Reply:
i was suggesting to just put the raw html message in there lol

a lot of people will miss the fact that’s the message content because it being b64’d is unusual, that’s why i pointed it out

Reply:
I can add a comment

Reply:
i don’t understand why you just don’t put the content there, what are you achieving 😭

Reply:
> i was suggesting to just put the raw html message in there lol

The raw html was kinda big, that's why I encoded it. My initial html has 346 lines which I was able to trim down a lot, maybe it's worth a try

Reply:
> > i was suggesting to just put the raw html message in there lol
> 
> The raw html was kinda big, that's why I encoded it. My initial html has 346 lines which I was able to trim down a lot, maybe it's worth a try

how much html do you need for a gdpr zip file

Reply:
just decoded the base64, this is really like nothing, just put the raw message in there

Reply:
then you can also use an f-string and remove the stupid replaces on domain and username

Reply:
> how much html do you need for a gdpr zip file

The designer I used just put all kinds of bullshit in it

Reply:
![image](https://user-images.githubusercontent.com/33896771/212393307-fc81f70a-4cb6-4ab8-b756-de58161dca31.png)
it's not that big i think

Reply:
yeah I know it's not that big